src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitSubscriber.java

```diff
@@ -106,85 +109,63 @@ public void init(@NotNull ConnectionManager connectionManager, @NotNull Subscrib
     }
 
     @Override
+    @Deprecated
     public void start() throws Exception {
-        ConnectionManager connectionManager = this.connectionManager.get();
-        if (connectionManager == null) {
-            throw new IllegalStateException("Subscriber is not initialized");
-        }
+        // Do nothing
+    }
 
+    @Override
+    public void addListener(ConfirmationMessageListener<T> messageListener) {
         try {
-            consumerMonitor.updateAndGet(monitor -> {
-                if (monitor == null) {
-                    try {
-                        monitor = connectionManager.basicConsume(
-                                queue,
-                                (consumeTag, delivery, confirmation) -> {
-                                    Timer processTimer = MESSAGE_PROCESS_DURATION_SECONDS
-                                            .labels(th2Pin, th2Type, queue)
-                                            .startTimer();
-                                    MESSAGE_SIZE_SUBSCRIBE_BYTES
-                                            .labels(th2Pin, th2Type, queue)
-                                            .inc(delivery.getBody().length);
-                                    try {
-                                        T value;
-                                        try {
-                                            value = valueFromBytes(delivery.getBody());
-                                        } catch (Exception e) {
-                                            LOGGER.error("Couldn't parse delivery. Confirm message received", e);
-                                            confirmation.confirm();
-                                            throw new IOException(
-                                                    String.format(
-                                                            "Can not extract value from bytes for envelope '%s', queue '%s', pin '%s'",
-                                                            delivery.getEnvelope(), queue, th2Pin
-                                                    ),
-                                                    e
-                                            );
-                                        }
-                                        handle(consumeTag, delivery, value, confirmation);
-                                    } finally {
-                                        processTimer.observeDuration();
-                                    }
-                                },
-                                this::canceled
-                        );
-                        LOGGER.info("Start listening queue name='{}'", queue);
-                    } catch (IOException e) {
-                        throw new IllegalStateException("Can not start subscribe to queue = " + queue, e);
-                    }
+            publicLock.lock();
```

Comment:
We should take the lock outside try-finally block. Because, it an error was thrown when we tried to take a lock we should not release the lock

Reply:
Which one exception do you mean?